src/autofix/autofix.ts

```diff
@@ -0,0 +1,318 @@
+	for (const [resourcePath, sourceFile] of sourceFiles) {
+		const newContent = applyFixes(checker, sourceFile, resourcePath, resources.get(resourcePath)!, context);
+		if (newContent) {
+			res.set(resourcePath, newContent);
+		}
+	}
+
+	return res;
+}
+
+function applyFixes(
+	checker: ts.TypeChecker, sourceFile: ts.SourceFile, resourcePath: ResourcePath,
+	resource: AutofixResource, context: LinterContext
+): string | undefined {
+	const {content} = resource;
+
+	// Group messages by id
+	const messagesById = new Map<MESSAGE, RawLintMessage[]>();
+	for (const msg of resource.messages) {
+		if (!messagesById.has(msg.id)) {
+			messagesById.set(msg.id, []);
+		}
+		messagesById.get(msg.id)!.push(msg);
+	}
+
+	const changeSet: ChangeSet[] = [];
+	let existingModuleDeclarations = new Map<ts.CallExpression, ExistingModuleDeclarationInfo>();
+	if (messagesById.has(MESSAGE.NO_GLOBALS)) {
+		existingModuleDeclarations = generateSolutionNoGlobals(
+			checker, sourceFile, content,
+			messagesById.get(MESSAGE.NO_GLOBALS) as RawLintMessage<MESSAGE.NO_GLOBALS>[],
+			changeSet, []);
+	}
+
+	for (const [defineCall, moduleDeclarationInfo] of existingModuleDeclarations) {
+		addDependencies(defineCall, moduleDeclarationInfo, changeSet, context.getMetadata(resourcePath));
+	}
+
+	if (changeSet.length === 0) {
+		// No modifications needed
+		return undefined;
+	}
+	return applyChanges(content, changeSet);
+}
+
+function addDependencies(
+	defineCall: ts.CallExpression, moduleDeclarationInfo: ExistingModuleDeclarationInfo,
+	changeSet: ChangeSet[], resourceMetadata: LintMetadata
+) {
+	const {moduleDeclaration, importRequests} = moduleDeclarationInfo;
+
+	const existingImportModules = resourceMetadata.transformedImports.get("sap.ui.define") ?? [];
+	const moduleWithoutIdentifier = existingImportModules.find((i) => !i.identifier);
+	const existingIdentifiersLength = moduleWithoutIdentifier ?
+			existingImportModules.indexOf(moduleWithoutIdentifier) :
+		existingImportModules.length ?? 0;
+
+	const imports = [...importRequests.keys()];
+	existingImportModules.forEach((existingImportModule) => {
+		const indexOf = imports.indexOf(existingImportModule.moduleName);
+		if (indexOf !== -1) {
+			imports.splice(indexOf, 1);
+			importRequests.get(existingImportModule.moduleName)!.identifier = existingImportModule.identifier;
+		}
+	});
+
+	const dependencies = imports.map((i) => `"${i}"`);
+	const identifiers = [
+		...existingImportModules.map((i) => i.identifier || getIdentifierForImport(i.moduleName))
+			.slice(existingIdentifiersLength),
+		...imports.map((i) => {
+			const identifier = getIdentifierForImport(i);
+			importRequests.get(i)!.identifier = identifier;
+			return identifier;
+		})];
+
+	if (moduleDeclaration.dependencies) {
+		const depsNode = defineCall.arguments[0];
+		const depElementNodes = depsNode && ts.isArrayLiteralExpression(depsNode) ? depsNode.elements : [];
+		const lastElement = depElementNodes[depElementNodes.length - 1];
```

Comment:
this fails if `depsNode` is empty, as there is no element at all. Please add more test fixtures to cover the non-straight-forward cases. Currently we only have two fixtures (+ the e2e test)

Reply:
This is not possible. Please take a look at line #231
This is the case where we are ensured that there are dependencies already and we take the last one

Reply:
Ah, thanks!